include/drivers/espi_emul.h

```diff
@@ -88,11 +99,13 @@ typedef struct espi_emul *(*emul_find_emul)(const struct device *dev,
 typedef int (*emul_trigger_event)(const struct device *dev,
 				  struct espi_event *evt);
 
-
 /** Definition of the eSPI device emulator API */
 struct emul_espi_device_api {
 	emul_espi_api_set_vw set_vw;
 	emul_espi_api_get_vw get_vw;
+#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
+	emul_espi_api_get_acpi_shm get_acpi_shm;
```

Comment:
Can we add write_lpc_request instead and return a proper value on the host side based on the opcode? Or do you think the current approach is better?

Reply:
Just to make sure I'm on the same page. It looks like the implementation of `read_lpc_request` for the NPCX chip is to basically just return the address of the shared memory. I thought about implementing that and basically only supporting the one op code for now (it would look similar to `host_subs_npcx.c::npcx_host_periph_read_request()` and handle `EACPI_GET_SHARED_MEMORY` only).

The `write_lpc_request` doesn't seem to access the shared memory (AFAICT), is there something I might be missing in how the emulator should handle the shared memory?

Reply:
You are right, it should be `read_lpc_request`, similar to the `host_subs_npcx.c::npcx_host_periph_read_request`.

Reply:
I'm looking at this some more and I think I might have confused myself :)

My understanding of the PR code as it is currently is that:
* I added a `.get_acpi_shm` function to the emulated eSPI host device. This is done in `emul_espi_host.c`.
* I then set both `.read_lpc_request` and `.write_lpc_request` for the emulated eSPI device in `espi_emul.c`. This means that the emulated eSPI device should now be able to call those functions: `espi_read_lpc_request` and `espi_write_lpc_request`.
* The implementation of the write function is a stub and will always return an error for now
* The implementation of the read will only handle the `EACPI_GET_SHARED_MEMORY` operation and will call the host function to directly get the shared memory address and cast it to a `uint32_t`.
* During the test, we can mock host interaction with the memory by also calling the same function to get the memory address and read/write to it.

Adding `read_lpc_request` to the host emulator might not make sense for all op codes (I'm not very familiar with eSPI, but that's my understanding). So instead I think that for the host emulator to have direct access functions to the specific components such as the shared memory makes the most sense.

One thing that I do think might come in handy (but I didn't add yet until we find a use for it) is to add a host API to get the shared memory directly. This would just replicate the typedef for `emul_espi_api_get_acpi_shm` in `espi_emul.h` but would allow global access such as `intptr_t shm = emul_espi_host_get_acpi_shm(espi_host);`.

Reply:
Regarding `emul_espi_host_get_acpi_shm`, I was about to suggest you to add this kind of function. I think it is better to have  access to the share memory via the emulated host, instead of calling functions from real eSPI API(which is of course also emulated).

I'm also not an eSPI expert, but dedicated host functions sounds good for me.

Regarding the bullets, my understanding of the PR is the same, except for the last point once the `emul_espi_host_get_acpi_shm` function is added.

Reply:
Sounds good, I'll make the changes soon. I also just thought of a good unit test that I think would work for this so I'll bundle that up in the same PR.